### PR TITLE
Avoid redundant bazel build

### PR DESCRIPTION
**Pull request by raulchen**
## What do these changes do?

Bazel will invalidate caches if the command-line options are different. This PR puts all needed `action_env`s in the `.bazelrc` file to avoid cache invalidation.

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -1,2 +1,4 @@
 # build config
 build --compilation_mode=opt
+build --action_env=PATH
```
Comment (raulchen) [inline .bazelrc]: @pcmoritz I think by passing `PATH` to bazel, #4334 is not needed any more. But I don't know how to verify that.

**Review by guoyuhong — APPROVED** [review]

**Comment by raulchen** [comment]
Java CI time reduced from 32min to 17min

File: .bazelrc (hunk 1 above)
Comment (pcmoritz) [reply .bazelrc]: It is still needed, otherwise we can't guarantee which python will be picked up if there are multiple ones.

**Review by pcmoritz — APPROVED** [review]
LGTM, but let's be careful to monitor that introducing PATH into the python build doesn't break things (i.e. different binaries might be picked up based on whether PATH is set or not).

File: .bazelrc (hunk 1 above)
Comment (raulchen) [reply .bazelrc]: ok, just noticed that the `build.sh` script takes a `-p` option.

**Pull request merged**